src/LibraryManager.Vsix/Shared/LibraryHelpers.cs

```diff
@@ -47,75 +47,151 @@ internal static class LibraryHelpers
 
         private static async Task RestoreAsync(IDictionary<string, Manifest> manifests, CancellationToken cancellationToken = default(CancellationToken))
         {
-            Logger.LogEvent(LibraryManager.Resources.Text.RestoringLibraries, LogLevel.Status);
+            Stopwatch sw = new Stopwatch();
+
+            Logger.LogEvent(LibraryManager.Resources.Text.RestoringLibraries, LogLevel.Task);
+            List<ILibraryInstallationResult> totalResults = new List<ILibraryInstallationResult>();
 
-            var sw = new Stopwatch();
             sw.Start();
-            int resultCount = 0;
-            bool hasErrors = false;
-            var telResult = new Dictionary<string, double>();
 
             foreach (KeyValuePair<string, Manifest> manifest in manifests)
             {
+                Project project = VsHelpers.DTE.Solution?.FindProjectItem(manifest.Key)?.ContainingProject;
+                Logger.LogEvent(string.Format("Restoring packages for {0}...", project.FullName), LogLevel.Operation);
+
                 IEnumerable<ILibraryInstallationResult> results = await RestoreLibrariesAsync(manifest.Value, cancellationToken).ConfigureAwait(false);
 
-                Project project = VsHelpers.DTE.Solution?.FindProjectItem(manifest.Key)?.ContainingProject;
                 AddFilesToProject(manifest.Key, project, results);
+                AddErrorsToErrorList(project?.Name, manifest.Key, results);
+
+                totalResults.AddRange(results);
+            }
+
+            sw.Stop();
+
+            Logger.LogEvent("Restoring libraries completed", LogLevel.Status);
 
-                var errorList = new ErrorList(project?.Name, manifest.Key);
-                hasErrors |= errorList.HandleErrors(results);
+            PostRestoreTelemetryData(totalResults, sw.Elapsed);
+            LogEventsSummary(totalResults, sw.Elapsed);
 
-                resultCount += results.Count();
+        }
 
-                foreach (ILibraryInstallationResult result in results.Where(r => r.Success))
+        private static void LogEventsSummary(List<ILibraryInstallationResult> totalResults, TimeSpan elapsedTime)
```

Comment:
Make it generic enough to take any operation type (Restore, Clean, Uninstall, Update), IEnumerable<ILibraryInstalltionResult> Move to the Logger implementation